dataRetriever/dataPool/proofsCache/proofsBucket.go

```diff
@@ -0,0 +1,44 @@
+package proofscache
+
+import "github.com/multiversx/mx-chain-core-go/data"
+
+type proofNonceBucket struct {
+	maxNonce      uint64
+	proofsByNonce []*proofNonceMapping
+	bucketSize    int
+}
+
+func newProofBucket(bucketSize int) *proofNonceBucket {
+	return &proofNonceBucket{
+		proofsByNonce: make([]*proofNonceMapping, 0),
+		bucketSize:    bucketSize,
+	}
+}
+
+func (p *proofNonceBucket) size() int {
+	return len(p.proofsByNonce)
+}
+
+func (p *proofNonceBucket) isFull() bool {
+	return len(p.proofsByNonce) >= p.bucketSize
+}
+
+func (p *proofNonceBucket) insertInNew(proof data.HeaderProofHandler) {
+	p.insert(proof)
+	p.maxNonce = proof.GetHeaderNonce()
+}
+
+func (p *proofNonceBucket) insertInExisting(proof data.HeaderProofHandler) {
+	p.insert(proof)
+
+	if proof.GetHeaderNonce() > p.maxNonce {
+		p.maxNonce = proof.GetHeaderNonce()
+	}
+}
```

Comment:
Can we merge them into a single `insert` function or does it break the logic? If it was done for the purpose of optimizing the nonce check - the `if proof.GetHeaderNonce() > p.maxNonce {` is just a quick comparison (should be negligible).

Reply:
Outdated comment, fixed upon Sorin's suggestion.